packages/client/lib/client/index.ts

```diff
@@ -426,11 +426,11 @@ export default class RedisClient<
         });
     }
 
-    async connect(): Promise<RedisClientType<M, F, S>> {
+    async connect() {
         // see comment in constructor
         this.#isolationPool ??= this.#initiateIsolationPool();
         await this.#socket.connect();
-        return this;
+        return this as unknown as RedisClientType<M, F, S>;
     }
```

Comment:
Looks good! I am not familiar enough with TS to know if there's any meaningful difference here, but I've seen that style in the past so sounds good.